[Bug] Cloudflare Workers deployment fails with "Completion token has already been consumed" [code: 100312]

> **Severity:** 🟠 High - Core feature is broken
> **Frequency:** Sometimes (30-70%)

<br />

## ❓ Problem

When deploying to Cloudflare Workers via GitHub Actions using `wrangler versions upload`, the deployment intermittently fails with the following error:
```sh
✘ [ERROR] A request to the Cloudflare API (/accounts/***/workers/scripts/blog/versions) failed.

Completion token has already been consumed [code: 100312]
```
This is caused by the workflow-level `cancel-in-progress: true` concurrency setting. When a new push triggers a new workflow run while a deployment is already in progress, GitHub Actions cancels the previous workflow — but by that point, Cloudflare may have already issued and consumed the completion token for that upload session, leaving it in an unrecoverable state for any subsequent attempt.

<br />

## ↪️ Steps to Reproduce

1. Push a commit to a non-`main` branch (e.g., `release/*` or `dependabot/**`)
2. Immediately push another commit to the same branch before the first deployment finishes
3. The first workflow run is cancelled due to `cancel-in-progress: true` at the workflow level
4. The second workflow run starts and `wrangler versions upload` fails with:
   "Completion token has already been consumed [code: 100312]"

<br />

## 🎯 Expected Behavior

Each deployment triggered by a push should complete independently without token conflicts.
When multiple pushes occur in quick succession, deployment jobs should be queued and executed sequentially rather than being cancelled mid-flight, ensuring that Cloudflare's completion token lifecycle is respected.

<br />

### ⚙️ Device Information

- Version: `v0.1.1`
- OS: GitHub Actions runner (ubuntu-latest)

### 🔗 Reference

- Affected PR: #35 

<img width="1517" height="694" alt="Image" src="https://github.com/user-attachments/assets/db3f6a20-4ded-45f4-8e9f-13eb9e435e90" />

### ➕ Additional Information

- Root cause:
  The `deploy-worker` job relied solely on the workflow-level concurrency group
  (`cancel-in-progress: true`). When a new workflow run cancelled an in-progress
  deployment, Wrangler's completion token had already been consumed on the
  Cloudflare side, causing the next run to fail.

- Proposed fix:
  Add a job-level concurrency group to `deploy-worker` with `cancel-in-progress: false`
  so that deployments are queued rather than interrupted mid-flight.

```sh
 concurrency:
    group: deploy-worker-${{ github.ref }}
    cancel-in-progress: false
```

---

### ✅ Checklist

- [x] I have searched for duplicate issues
- [x] I have tested on the latest version
- [ ] I have checked the browser console for errors

<br />

> ⚡ Generated by **GitHub Actions**

--- a/.github/workflows/deployment.yml
+++ b/.github/workflows/deployment.yml
@@ -21,6 +21,9 @@ jobs:
     name: Deploy to Cloudflare Workers
     runs-on: ubuntu-latest
     timeout-minutes: 10
+    concurrency:
+      group: deploy-worker-${{ github.ref }}
+      cancel-in-progress: false
     permissions:
       contents: read
     outputs:

